python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -217,6 +224,30 @@ def __init__(
                 "typ.buffer_size_bytes must be at least MIN_BUFFER_SIZE "
                 f"({MIN_BUFFER_SIZE} bytes)"
             )
+        if typ.buffer_size_bytes > GRPC_LIMIT:
+            raise ValueError(
+                "The object written to the channel must have a size less than or "
+                f"equal to the max gRPC payload size ({GRPC_LIMIT} bytes)"
+            )
+
+        # For now, all readers must be on the same node. Note that the writer can still
+        # be on a different node than the readers though.
+        # TODO(jhumphri): Allow different readers for the same channel to be on
+        # different nodes.
+        prev_reader_node = None
+        prev_reader = None
+        for reader in readers:
+            node = _get_reader_node_id(self, reader)
+            if prev_reader_node is None:
+                prev_reader_node = node
+            else:
+                if prev_reader_node != node:
+                    raise ValueError(
+                        "All reader actors must be on the same node. Actor "
+                        f"{prev_reader} is on node {prev_reader_node} while actor "
+                        f"{reader} is on node {node}"
+                    )
+            prev_reader = reader
```

Comment:
Doesn't really matter, but I don't think this line is necessary.

Reply:
I print it in the error message

Reply:
Oh my mistake, I read this as `prev_reader_node`!